generator/README.md

```diff
@@ -0,0 +1,1186 @@
+```
+
+Take for an instance the enum class KeypressMode
+ 
+```xml
+<enum name="KeypressMode" since="3.0">
+    <description>Enumeration listing possible keyboard events.</description>
+    <element name="SINGLE_KEYPRESS">
+        <description>Each keypress is individually sent as the user presses the keyboard keys.</description>
+    </element>
+    <element name="QUEUE_KEYPRESSES">
+        <description>The keypresses are queued and a string is eventually sent once the user chooses to submit their entry.</description>
+    </element>
+    <element name="RESEND_CURRENT_ENTRY">
+        <description>The keypresses are queue and a string is sent each time the user presses a keyboard key; the string contains the entire current entry.</description>
+    </element>
+</enum>
+```
+
+In the following example 
+
+```objc
+extern SDLKeypressMode const SDLKeypressModeSingleKeypress;
+```
+
+The **SDLKeypressModeSingleKeypress** enum is seen by the compiler as:
+
+```objc
+extern NSString* const SDLKeypressModeSingleKeypress;
+```
+
+and **SDLKeypressModeSingleKeypress** itself must be implemented in the correspondent SDLKeypressMode.m file as a string:
+
+```objc
+SDLKeypressMode const SDLKeypressModeSingleKeypress = @"SINGLE_KEYPRESS";
+```
+
+## Structures
+
+Structures in **sdl_ios**  implemented as classes derived from the parent class SDLRPCStruct with all parameters implemented as "@property". Let us take for an instance the **DeviceInfo** structure. In the XML it is declared as following:
+
+```xml
+<struct name="DeviceInfo" since="3.0">
+        <description>Various information about connecting device.</description>        
+        <param name="hardware" type="String"  minlength="0" maxlength="500" mandatory="false">
+            <description>Device model</description>
+        </param>
+        <param name="firmwareRev" type="String" minlength="0" maxlength="500" mandatory="false">
+            <description>Device firmware revision</description>
+        </param>
+        <param name="os" type="String" minlength="0" maxlength="500" mandatory="false">
+            <description>Device OS</description>
+        </param>
+        <param name="osVersion" type="String" minlength="0" maxlength="500" mandatory="false">
+            <description>Device OS version</description>
+        </param>
+        <param name="carrier" type="String" minlength="0" maxlength="500" mandatory="false">
+            <description>Device mobile carrier (if applicable)</description>
+        </param>
+        <param name="maxNumberRFCOMMPorts" type="Integer" minvalue="0" maxvalue="100" mandatory="false">
+            <description>Omitted if connected not via BT.</description>
+        </param>        
+ </struct>
+ ```
+ 
+ *Please note that all params declared as mandatory="false" and there is one init method with all the params in the generated file. The method* ```+ (instancetype)currentDevice;``` *comes from the old manually made implementation*
```

Comment:
This is unnecessary since the generator won't generate it.

Reply:
@joeljfischer done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/abb19f31a42d7d41122109e8a4993b7a4fb266cb